src/DslServices.h

```diff
@@ -2261,6 +2266,11 @@ namespace DSL {
          */
         FILE* m_debugLogFileHandle;
 
+        /**
+        * @brief Shared pointer to the spdlog logger instance for logging.
+        */
+        std::shared_ptr<spdlog::logger> m_spdLogger;
```

Comment:
You might want to consider using a raw pointer like:
spdlog::logger * m_spdLogger;

The reason I'm mentioning this is that I'm assuming that you are initializing and creating ownership in ACE and passing a reference to DSL. 

The shared pointer in ACE will automatically handle the memory de-allocation of the original pointer. 

You can simply just call the setup like this:
`SetSpdLogger(logger.get())`

Reply:
Thanks Kevin. I change logger management in DSL from shared_ptr copy to raw pointer reference. I tested it and it seems to be working well.

The logger instance is created and managed exclusively by ACE, and DSL only needs a reference to it for logging purposes. DSL does not take ownership or manage the logger's lifetime, aligning with the intended design where ACE handles all logger management.